src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/IntellisenseQuickInfoBuilder.cs

```diff
@@ -114,19 +111,42 @@ internal static class IntellisenseQuickInfoBuilder
                 var tabSize = document.Project.Solution.Workspace.Options.GetOption(FormattingOptions.TabSize, document.Project.Language);
                 var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
                 var spans = IndentationHelper.GetSpansWithAlignedIndentation(text, classifiedSpanList.ToImmutableArray(), tabSize);
-                var textRuns = spans.Select(s => new ClassifiedTextRun(s.ClassificationType, snapshot.GetText(s.TextSpan.ToSpan()), ClassifiedTextRunStyle.UseClassificationFont));
+                var textRuns = spans.Select(s => new ClassifiedTextRun(s.ClassificationType, text.GetSubText(s.TextSpan).ToString(), ClassifiedTextRunStyle.UseClassificationFont));
 
                 if (textRuns.Any())
                 {
                     elements.Add(new ClassifiedTextElement(textRuns));
                 }
             }
 
-            var content = new ContainerElement(
+            return new ContainerElement(
                                 ContainerElementStyle.Stacked | ContainerElementStyle.VerticalPadding,
                                 elements);
+        }
+
+        internal static async Task<IntellisenseQuickInfoItem> BuildItemAsync(ITrackingSpan trackingSpan,
+            CodeAnalysisQuickInfoItem quickInfoItem,
+            Document document,
+            IThreadingContext threadingContext,
+            Lazy<IStreamingFindUsagesPresenter> streamingPresenter,
+            CancellationToken cancellationToken)
+        {
+            var content = await BuildInteractiveContentAsync(quickInfoItem, document, threadingContext, streamingPresenter, cancellationToken).ConfigureAwait(false);
 
             return new IntellisenseQuickInfoItem(trackingSpan, content);
         }
+
+        /// <summary>
+        /// Builds the classified hover content without navigation actions and requiring
+        /// an instance of <see cref="IStreamingFindUsagesPresenter"/>
+        /// TODO - This can be removed once LSP supports colorization in markupcontent
+        /// https://devdiv.visualstudio.com/DevDiv/_workitems/edit/918138
+        /// </summary>
+        internal static Task<ContainerElement> BuildContentWithoutNavigationActionsAsync(CodeAnalysisQuickInfoItem quickInfoItem,
+            Document document,
+            CancellationToken cancellationToken)
+        {
+            return BuildInteractiveContentAsync(quickInfoItem, document, threadingContext: null, streamingPresenter: null, cancellationToken);
```

Comment:
interesting.  dont' love it.  why don't we actually have tehse guys to pass along?

Reply:
seems like a roundabout way to say "do not add hyperlinks."

Reply:
ah, n/m  i see that's how you're directly exposing it (BuildContentWithoutNavigationActionsAsync)